modules/web/src/app/node-data/basic/provider/kubevirt/component.ts

```diff
@@ -662,13 +662,18 @@ export class KubeVirtBasicNodeDataComponent
   private _init(): void {
     if (this._initialData) {
       this.form.get(Controls.Memory).setValue(parseInt(this._initialData.memory) || this._defaultMemory);
-      this.form.get(Controls.CPUs).setValue(parseInt(this._initialData.cpus) || this._defaultCPUs);
       this.form.get(Controls.PrimaryDiskSize).setValue(this._initialData.primaryDiskSize);
       this.form.get(Controls.PrimaryDiskOSImage).setValue(this._initialData.primaryDiskOSImage);
       this.form.get(Controls.NodeAffinityPreset).setValue(this._initialData.nodeAffinityPreset?.Type);
       this.form.get(Controls.NodeAffinityPresetKey).setValue(this._initialData.nodeAffinityPreset?.Key);
       this.nodeAffinityPresetValues = this._initialData.nodeAffinityPreset?.Values || [];
 
+      if (!_.isEmpty(this._initialData.vcpus)) {
+        this.form.get(Controls.CPUs).setValue(this._initialData.vcpus.cores);
+      } else {
+        this.form.get(Controls.CPUs).setValue(parseInt(this._initialData.cpus) || this._defaultCPUs);
+      }
+
```

Comment:
This looks rather wrong and hackish. UI should deal with vCPUs and CPUs dedicatedly as individual fields. This should have a 1-1 relation with the machine deployment.

Reply:
@soer3n for the UI I will speak with @ahmadhamzh if he can assist us even though I am sure that we don't need any UI changes. Let's discuss later on but I believe UI changes can be reverted